pkg/antctl/raw/clusterset/command.go

```diff
@@ -0,0 +1,138 @@
+
+	antreamcscheme "antrea.io/antrea/multicluster/pkg/client/clientset/versioned/scheme"
+
+	k8sscheme "k8s.io/client-go/kubernetes/scheme"
+
+	mcsscheme "sigs.k8s.io/mcs-api/pkg/client/clientset/versioned/scheme"
+
+	"antrea.io/antrea/pkg/antctl/raw"
+
+	multiclusterv1alpha1 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha1"
+)
+
+var (
+	Command *cobra.Command
+	option  = &struct {
+		namespace string
+	}{}
+)
+
+func init() {
+	Command = &cobra.Command{
+		Use:     "clusterset",
+		Aliases: []string{"clustersets"},
+		Short:   "Print multicluster clustersets",
+		Args:    cobra.MaximumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			var err error
+			if len(args) != 0 {
+				err = runE(cmd, args[0])
+			} else {
+				err = runE(cmd, "")
+			}
+			if err != nil {
+				fmt.Println(fmt.Errorf("please input the correct numbers of argumrnts here"))
+			}
+		},
+	}
+	Command.Flags().StringVarP(&option.namespace, "namespace", "n", "", "namespace of the clusterset")
+}
+
+func runE(cmd *cobra.Command, clustersetName string) error {
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return err
+	}
+	kubeconfig.GroupVersion = &schema.GroupVersion{Group: "", Version: ""}
+	restconfigTmpl := rest.CopyConfig(kubeconfig)
+	raw.SetupKubeconfig(restconfigTmpl)
+	if server, err := Command.Flags().GetString("server"); err != nil {
+		kubeconfig.Host = server
+	}
+
+	scheme := runtime.NewScheme()
+	err = mcsscheme.AddToScheme(scheme)
+	if err != nil {
+		return err
+	}
+	err = antreamcscheme.AddToScheme(scheme)
+	if err != nil {
+		return err
+	}
+	err = k8sscheme.AddToScheme(scheme)
+	if err != nil {
+		return err
+	}
+
+	k8sClient, err := client.New(kubeconfig, client.Options{Scheme: scheme})
+	if err != nil {
+		return err
+	}
+
+	clusterSetList := &multiclusterv1alpha1.ClusterSetList{}
+	if option.namespace == "" {
+		err = k8sClient.List(context.Background(), clusterSetList, &client.ListOptions{Namespace: metav1.NamespaceAll})
+	} else {
+		err = k8sClient.List(context.Background(), clusterSetList, &client.ListOptions{Namespace: option.namespace})
+	}
+	if err != nil {
+		return fmt.Errorf("failed to list Clustersets: %w", err)
+	}
+
+	var clusterSets []multiclusterv1alpha1.ClusterSet
+
+	if clustersetName != "" {
+		for _, clusterSet := range clusterSetList.Items {
+			if clusterSet.Name == clustersetName {
+				clusterSets = append(clusterSets, clusterSet)
+			}
+		}
+	} else {
+		clusterSets = clusterSetList.Items
+	}
+
+	fmt.Println(output(clusterSets))
+	return nil
+}
+
+func output(clusterSets []multiclusterv1alpha1.ClusterSet) string {
+	var output strings.Builder
+	formatter := "%-25s%-25s%-28s%-20s%-40s%-50s%-10s\n"
+	output.Write([]byte(fmt.Sprintf(formatter, "CLUSTERID", "NAMESPACE", "TYPE", "STATUS", "LASTTRANSACTIONTIME", "MESSAGE", "REASON")))
+	for _, clusterSet := range clusterSets {
+		for _, r := range clusterSet.Status.ClusterStatuses {
+			for _, condition := range r.Conditions {
```

Comment:
what's the basic information in ClusterSet spec?

Reply:
Can you add a column named 'ClUSTERSETID', otherwise the output only shows the overall status and user is unable to tell which ClusterSet it belongs to:
```
CLUSTERID                NAMESPACE                TYPE                        STATUS              REASON
test-cluster-east        docs                     Ready                       Unknown             NeverConnected
test-cluster-east        docs                     ImportsResources            False               NeverConnected
test-cluster-west        docs                     Ready                       True                Connected
test-cluster-west        docs                     ImportsResources            True                ElectedLeader
test-cluster-north       docs                     Ready                       Unknown             NeverConnected
test-cluster-north       docs                     ImportsResources            False               NeverConnected
```

@aravindakidambi  any suggestion here?

Reply:
Could we do some thing like this:
1. Command to show summary of the ClusterSet
2. Command to show details of a ClusterSet, customer needs to provide a cluster-set-id to see the details
The above command seems more like #2 for me

For #1, we should just have these columns
`CLUSTERSETID  TOTAL_CLUSTERS  NUM_READY_CLUSTERS OVERALL_CONDITION REASON
test-clusterset-1    4                            3                                         Ready=True                  
test-cluster-2         2                            0                                         Ready=Unknown          NoReadyCluster`